components/scream/src/share/field/field_alloc_prop.cpp

```diff
@@ -85,16 +85,35 @@ subview (const int idim, const int k) const {
   return props;
 }
 
-int FieldAllocProp::get_alloc_size () const {
-  EKAT_REQUIRE_MSG(is_committed(),
-      "Error! You cannot query the allocation properties until they have been committed.");
-  return m_alloc_size;
+void FieldAllocProp::request_allocation (const int scalar_size, const int pack_size) {
+  using ekat::ScalarTraits;
+  using namespace ekat::error;
+
+  ekat::error::runtime_check(!m_committed, "Error! Cannot change allocation properties after they have been commited.\n");
+
+  const int vts = scalar_size*pack_size;
+  if (m_scalar_type_size==0) {
+    // This is the first time we receive a request. Set the scalar type properties
+    m_scalar_type_size = scalar_size;
+  } else {
+    // Make sure the new scalar_type coincides with the one already stored (check name and size)
+    runtime_check(scalar_size==m_scalar_type_size,
```

Comment:
Is this the same function as in line 92?

Reply:
Ah, yes. I'll remove the using directive, and do fully qualified namespace call.

Reply:
Btw, we have both `EKAT_REQUIRE_MSG` (a macro) as well as `runtime_check` (a function) in ekat. If the check fails, the former throws an exception, while the latter aborts (finalizing kokkos and MPI, which will likely end up causing kokkos errors when views are deallocated at exit, after kokkos finalization). Perhaps at some point we can decide we're ok with just one, and remove the other altogether. I'm for keeping the exception macro and purging the abort fcn.